src/platform/win32bigfilesystem.cpp

```diff
@@ -26,146 +26,142 @@ using rts::FourCC;
+                file_pos = be32toh(file_pos);
 
-    ArchivedFileInfo *info = new ArchivedFileInfo;
+                info->size = file_size;
+                info->position = file_pos;
+                info->archive_name = filename;
 
-    // Process each file info found in the Big file header.
-    for (unsigned int i = 0; i < file_count; ++i) {
-        int32_t file_size = 0;
-        int32_t file_pos = 0;
-        char namebuf[BIG_PATH_MAX];
+                int strlen = 0;
+                char *putp = namebuf;
 
-        // Read file size and position in the Big into host integer format.
-        file->Read(&file_pos, sizeof(file_pos));
-        file->Read(&file_size, sizeof(file_size));
-        file_size = be32toh(file_size);
-        file_pos = be32toh(file_pos);
+                for (; strlen < BIG_PATH_MAX; ++strlen) {
+                    file->Read(putp, sizeof(*putp));
 
-        info->size = file_size;
-        info->position = file_pos;
-        info->archive_name = filename;
+                    if (*putp == '\0') {
+                        break;
+                    }
 
-        int strlen = 0;
-        char *putp = namebuf;
+                    ++putp;
+                }
 
-        for (; strlen < BIG_PATH_MAX; ++strlen) {
-            file->Read(putp, sizeof(*putp));
+                captainslog_relassert(
+                    strlen < BIG_PATH_MAX, 0xDEAD0002, "Filename string in BIG file header not null terminated");
 
-            if (*putp == '\0') {
-                break;
-            }
+                // Find the start of the file name
+                int name_start = strlen;
 
-            ++putp;
-        }
+                for (; name_start >= 0; --name_start) {
+                    if (namebuf[name_start] == '\\' || namebuf[name_start] == '/') {
+                        break;
+                    }
+                }
 
-        captainslog_relassert(strlen < BIG_PATH_MAX, 0xDEAD0002, "Filename string in BIG file header not null terminated");
+                // Store the file name in the info struct and then null first char so we
+                // can recover the rest of the path.
+                info->file_name = &namebuf[name_start + 1];
+                info->file_name.To_Lower();
+                // captainslog_trace("Base name is '%s'.", &namebuf[name_start + 1]);
 
-        // captainslog_trace("Recovered a file path of '%s' with size '%d' and position '%d'.", namebuf, file_size,
-        // file_pos);
+                namebuf[name_start + 1] = '\0';
 
-        // Find the start of the file name
-        int name_start = strlen;
+                // captainslog_trace("Path is '%s'.", namebuf);
 
-        for (; name_start >= 0; --name_start) {
-            if (namebuf[name_start] == '\\' || namebuf[name_start] == '/') {
-                break;
+                Utf8String file_path = namebuf;
+                file_path += info->file_name;
+                big->Add_File(namebuf, info);
```

Comment:
This looks odd. Why is `file_path` unused?

Reply:
Seems to be unused leftover, it's in the code but its value is not used anywhere.